packages/cli/src/deprecation/deprecation.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Service } from 'typedi';
+
+import { Logger } from '@/logging/logger.service';
+
+type Deprecation = {
+	env: string;
+	message: string;
+	inUse?: boolean;
+	checkValue?: (value: string) => boolean;
+};
+
+const SAFE_TO_REMOVE = 'This can be safely removed.';
+
+@Service()
+export class DeprecationService {
+	private readonly deprecations: Deprecation[] = [
+		{ env: 'N8N_BINARY_DATA_TTL', message: SAFE_TO_REMOVE },
+		{ env: 'N8N_PERSISTED_BINARY_DATA_TTL', message: SAFE_TO_REMOVE },
+		{ env: 'EXECUTIONS_DATA_PRUNE_TIMEOUT', message: SAFE_TO_REMOVE },
+		{
+			env: 'N8N_BINARY_DATA_MODE',
+			message: '`default` is deprecated. Please switch to `filesystem` mode.',
+			checkValue: (value: string) => value === 'default',
+		},
+		{ env: 'N8N_CONFIG_FILES', message: 'Please use .env files or *_FILE env vars instead.' },
+		{
+			env: 'DB_TYPE',
+			message: 'MySQL and MariaDB are deprecated. Please migrate to PostgreSQL.',
+			checkValue: (value: string) => ['mysqldb', 'mariadb'].includes(value),
+		},
+		{ env: 'N8N_PATH', message: 'Please use N8N_BASE_URL instead.' },
+		{ env: 'N8N_SKIP_WEBHOOK_DEREGISTRATION_SHUTDOWN', message: SAFE_TO_REMOVE },
```

Comment:
```suggestion
		{ env: 'N8N_SKIP_WEBHOOK_DEREGISTRATION_SHUTDOWN', message: 'n8n no longer deregisters webhooks at startup and shutdown, in main and queue mode. This can be safely removed.' },
```

Reply:
Here I intentionally excluded the explanation because
- this has already been deprecated already for a while
- in the others we don't go into this level of detail (or any)
- I really wonder if this makes sense to a user with no context

Let me know otherwise!

Reply:
Personally I'd like to add as much as possible (relevant) context. I don't think leaving it out because it looks different from others is a good reason to leave it out. IMHO it's fine if the users don't necessarily understand it, as the course of action is in the end. To make it clearer, maybe we can omit the `, in main and queue mode.`, as there are no other modes?